### ISSUE TITLE
Feature Request: Developer Experience

### ISSUE THREAD

**Issue by benc-uk** [microsoft/azure-container-apps#182]
**Is your feature request related to a problem? Please describe**.  
Having spent a few days deploying several apps to Azure Container Apps (ACA) I've found many gaps in the usability and general developer experience 

**Describe the solution you'd like**.    

Here is a list of some of the things I bumped up against 

- No way to see status of containers in the scale/replica set, if they are running or not. And overall heath / running status is very buried and gives almost no info
- No way to view or query; events, restarts, terminations etc.
- No way to see stdout & logs without using Log Analytics & KQL plus waiting 5+ minutes. This is a horrible experience, it's extremely hard to read, you see historical data when you don't want to, and timestamps don't always align.
- Readiness & liveness probes, again no way to see or query any status; if they are failing or working. Who knows?!
- Container Instances allowed you get a terminal into the container directly from the portal, great for debugging. No such option here, even from the CLI.
- API specs are missing for `Microsoft.App/containerApps@2022-01-01-preview` meaning no type hinting in Bicep and many warnings.
- No way to see or manage assigned managed identities from the portal

**Describe alternatives you've considered**.   
Many of these features such as live stdout, and details of restarts etc are available in Container Instances

**Additional context**.

**Comment by anthonychu** [comment]
Thanks for the feedback!

> No way to see status of containers in the scale/replica set, if they are running or not. And overall heath / running status is very buried and gives almost no info

We're working on an `az containerapp replica list` command which will help with this somewhat. ETA this month. For now, you need to use `az rest -u "/subscriptions/.../resourceGroups/.../providers/Microsoft.App/containerApps/.../revisions/.../replicas?api-version=2022-01-01-preview"`

There are also plans to add more details to enhance the information displayed at the app, revision, and replica levels. Please share more specifics about what you're looking for so we can take those into account.

A replicas view will be coming to the portal as well. Work hasn't started so no ETA yet.

> No way to view or query; events, restarts, terminations etc.

I think we have something in the works for this. @BigMorty

> No way to see stdout & logs without using Log Analytics & KQL plus waiting 5+ minutes. This is a horrible experience, it's extremely hard to read, you see historical data when you don't want to, and timestamps don't always align.

Log streaming coming soon to CLI and portal.

> Readiness & liveness probes, again no way to see or query any status; if they are failing or working. Who knows?!

@BigMorty 

> Container Instances allowed you get a terminal into the container directly from the portal, great for debugging. No such option here, even from the CLI.

Coming soon!

> API specs are missing for Microsoft.App/containerApps@2022-01-01-preview meaning no type hinting in Bicep and many warnings.

@ruslany Can you help with this?

> No way to see or manage assigned managed identities from the portal

We hit a bug the Identity blade is currently hidden. It'll be there in a couple of weeks.

**Comment by BigMorty** [comment]
Yes, we are currently working on a way to see events like restarts, health probe failures and more. Initial solution will use Log Analytics for these events and soon after we will have an API, Portal UX, and CLI command to see these.

**Comment by benc-uk** [comment]
Thanks for listening and taking a look, I figured many of these would be on the roadmap, and good to see they are - but I also thought - if I don't raise them with the team, I have no right to complain!

**Comment by sshquack** [comment]
I ran into some confusing problem with tags that was hard to troubleshoot. So I'm adding my feedback here:
- As part of ACA events, it would be helpful to see things like revision creation along with restarts, termination and scaling.
- When running `az containerapp update`, the CLI should clearly display if a new revision was created or not. Currently the CLI dumps out a JSON with no indication of if a new revision was created or not. So I kept running it a few times and later realized that the `revision list` command does not reflect my `update` because there was no change.
- The CLI and portal should always show the current image tag (`<REGISTRY>:<TAG>`) and the image ID (`<SHA:HASH>`). The reason for this is there is no way in ACR to prevent tags from being overwritten. This means a tag `v1` could be pointing to `HASH-123` or `HASH-XYZ`. This will help diagnose any mismatches in the running containers.
